enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,284 @@
+
+
+```Go
+// EtcdBackupStatus represents the status of the EtcdBackup CRD.
+type EtcdBackupStatus struct {
+
+    // Conditions represents the observations of the EtcdBackup's current state.
+    // TODO: Identify different condition types/reasons that will be needed.
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+    
+    // ObservedGeneration is the most recent generation observed for this
+    // EtcdBackup. It corresponds to the EtcdBackup's .metadata.generation that the condition was set based upon.
+    // This lets us know whether the latest condition is stale.
+    // +optional
+    ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+    // TODO: How do we track the state of the current/last backup?
+        // Would condition states be enough? e.g Created/In-progress/Complete/Failed/Unknown
+    // Track metadata on last successful backup?
+        // node name, path, size, timestamp
+    // With backup retention do we also track the last backup removed?
+}
+```
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Backup controller
+
+A backup controller in the cluster-etcd-operator will reconcile the CRD to execute the backup schedule and report the status of ongoing backups and failures to the `EtcdBackup` status.
+
+There are different options to explore on how we want to execute saving the backup snapshot and any other required metadata. As well as how we enforce the schedule.
+
+#### Execution method
+
+Create a backup pod that runs the backup script on a designated master node to save the backup file on the node's host filesystem:
+- The CEO already has an existing [upgradebackupcontroller][upgradebackupcontroller] that deploys a [backup pod][backup pod] that runs the [backup script][backup script].
+  - It may be simpler to reuse that but we may need to modify the backup script to save or append additional metadata.
+  - Making changes to a bash script without unit tests would make it harder to maintain.
+
+As an alternative we could implement an equivalent Go cmd for the backup pod to execute a variant of the backup script.
+- The etcd Go client provides a [`snapshot save()`](https://pkg.go.dev/go.etcd.io/etcd/client/v3@v3.5.7/snapshot) util that can be used for this method.
+- We would be maintaining two potentially different backup procedures this way. The backup script won't be replaced by the `EtcdBackup` API since it can still be used in a quorum-loss scenario where the API server is down.
+
+Since it is an implementation detail with no API impact we can start with utilizing the backup script for simplicity and codify it later.
+
+#### Backup schedule
+
+To enforce the specified schedule we can utilize [CronJobs](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) to run backup pods.
+
+**TODO:** If load balancing between control-plane nodes is a requirement then it needs to be determined how we can achieve that with the Job or Pod spec.
```

Comment:
actually, that's an interesting issue we wouldn't have with a single mounted PVC. 
Let's say I have RetentionCount = 3, how do we know where to find and delete the respective snapshot on which CP node?